src/librustc_interface/passes.rs

```diff
@@ -428,6 +455,16 @@ fn configure_and_expand_inner<'a>(
         }
     });
 
+    // // Needs to go *after* expansion, to be able to check the results of macro expansion.
+    // sess.time("complete_gated_feature_checking", || {
+    //     rustc_ast_passes::feature_gate::check_crate(
+    //         &krate,
+    //         &sess.parse_sess,
+    //         &sess.features_untracked(),
+    //         sess.opts.unstable_features,
+    //     );
+    // });
```

Comment:
I think moving this after `sess.parse_sess.buffered_lints.with_lock(|buffered_lints| {` will prevent the errors of this being emitted.

Reply:
Ah, I see it now. Thanks!

IIUC, `resolver.resolve_crate()` is expected to possibly add to the `lint_buffer` (possibly `check_crate()` too), so one resolution may be to copy the section under the comment `// Add all buffered lints...to the Session` after `resolve_crate` and `check_crate`.

All of the tests seemed to pass, maybe because they are not concerned with lint warnings (and that probably means there should be at least one more test to guard against this kind of change).